### ISSUE TITLE
Cluster-init prevented by server_url is set in config.yaml

### ISSUE THREAD

**Issue by Moep90** [rancher/k3os#379]
**Version (k3OS / kernel)**
```
$ k3os --version
k3os version v0.9.0

$ uname --kernel-release --kernel-version
5.0.0-37-generic #40~18.04.1 SMP Wed Jan 15 04:09:29 UTC 2020
```

**Architecture**
```
$ uname --machine
x86_64
```

**Describe the bug**

According to the documentation this will just be used by an `agent`. But if I want to bootstrap multi-master setup, it will try to connect to the givven server-urls as well.

> ### `k3os.server_url`
>
> The URL of the k3s server to join as an agent.
>
> Example
> ```yaml
> k3os:
>   server_url: https://myserver:6443
> ```

**To Reproduce**

Set server_url on a not-yet bootstrapped server.
```
k3os:
  server_url: https://myserver:6443
```

**Expected behavior**

If I want to bootstrap a server, it should use the URL just within the kubeconfig.
We maybe need to introduce a `cluster-init` flag. 

**Actual behavior**

The server tries to connect itself to the givven `server_url` which will prevent it from bootstrap.

**Comment by dweomer** [comment]
@Moep90 this is a documentation issue in that the [readme entry for `server_url`](https://github.com/rancher/k3os#k3osserver_url) has implied side-effects:

> ### `k3os.server_url`
> 
> The URL of the k3s server to join as an agent.
> 
> Example
> ```yaml
> k3os:
>   server_url: https://myserver:6443
> ```

Specifically:

> "The URL of the k3s server to join as an agent"

Is trying to say that when specifying a `server_url` k3s will assume that you want to run the instance as an **agent** unless you override `k3s_args` to specifically indicate that it should be a **server**.

**Comment by dweomer** [comment]
Additionally, I think that it is always considered an error to point a server to itself via the `server_url` config.

**Comment by Moep90** [comment]
> @Moep90 this is a documentation issue in that the [readme entry for `server_url`](https://github.com/rancher/k3os#k3osserver_url) has implied side-effects:
> 
> > ### `k3os.server_url`
> > The URL of the k3s server to join as an agent.
> > Example
> > ```yaml
> > k3os:
> >   server_url: https://myserver:6443
> > ```
> 
> Specifically:
> 
> > "The URL of the k3s server to join as an agent"
> 
> Is trying to say that when specifying a `server_url` k3s will assume that you want to run the instance as an **agent** unless you override `k3s_args` to specifically indicate that it should be a **server**.

Good point, but I'm rotation between k3s and k3os a lot to figure my problems out (eventually).

But here, it say's `server` in general: 
https://rancher.com/docs/k3s/latest/en/installation/install-options/

>    --server value, -s value                   (experimental/cluster) Server to connect to, used to join a cluster [$K3S_URL]

Maybe `server_url`shouldn't be considered when `k3os.args:` contains `server`?
But anyway this is not a major problem in my opinion.
What about the consideration to extend the yaml and split k3os mode and args, like:
```yaml
k3os.args: 
  - server
```

```yaml
k3os:
  role: master / worker / other future implemented roles
  args: 
    - "-v=3"
    - "--node-name={{ inventory_hostname }}"
    - "--server=https://{{ k3s_master.url }}:6443"
    - "--token={{ k3s_master.token }}"
```

**Comment by dweomer** [comment]
> But anyway this is not a major problem in my opinion.
> What about the consideration to extend the yaml and split k3os mode and args, like:
> 
> ```yaml
> k3os.args: 
>   - server
> ```
> 
> ```yaml
> k3os:
>   role: master / worker / other future implemented roles
>   args: 
>     - "-v=3"
>     - "--node-name={{ inventory_hostname }}"
>     - "--server=https://{{ k3s_master.url }}:6443"
>     - "--token={{ k3s_master.token }}"
> ```

I like this idea @Moep90

**Comment by danielo515** [comment]
> > "The URL of the k3s server to join as an agent"
> 
> Is trying to say that when specifying a `server_url` k3s will assume that you want to run the instance as an **agent** unless you override `k3s_args` to specifically indicate that it should be a **server**.

This is exactly what I understood, but in my case the server is trying to configure itself as a server and bind to that hostname (which is already pick by the expected master server).

Here is my log:

```
time="2020-10-31T12:21:09.100955246Z" level=info msg="Starting k3s agent v1.19.2+k3s1 (d38505b1)"
time="2020-10-31T12:21:09.101366389Z" level=info msg="module overlay was already loaded"
time="2020-10-31T12:21:09.101398137Z" level=info msg="module nf_conntrack was already loaded"
time="2020-10-31T12:21:09.101424927Z" level=info msg="module br_netfilter was already loaded"
time="2020-10-31T12:21:09.102074737Z" level=info msg="Running load balancer 127.0.0.1:34291 -> [kube01:6443]"
time="2020-10-31T12:21:09.293093790Z" level=error msg="Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
time="2020-10-31T12:21:14.391541894Z" level=error msg="Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
time="2020-10-31T12:21:19.481997589Z" level=error msg="Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
time="2020-10-31T12:21:24.565289968Z" level=error msg="Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
time="2020-10-31T12:21:29.644283689Z" level=error msg="Node password rejected, duplicate hostname or contents of '/etc/rancher/node/password' may not match server node-passwd entry, try enabling a unique node name with the --with-node-id flag"
```

I am looking for examples of how to configure a node as agent by customising an iso, but I can not find any clear reference to it.
